storage/series/series.go

```diff
@@ -401,26 +401,33 @@ func (s *dbSeries) bufferDrained(newBlock block.DatabaseBlock) {
 	// lock already. Executing the drain method occurs during a write if the
 	// buffer needs to drain or if tick is called and series explicitly asks
 	// the buffer to drain ready buckets.
-	s.mergeBlockWithLock(s.blocks, newBlock)
+	iOpts := s.opts.InstrumentOptions()
+	scope := iOpts.MetricsScope().SubScope("series-buffer-drain")
+	err := s.mergeBlockWithLock(newBlock)
+	if err != nil {
+		iOpts.Logger().WithFields(
+			xlog.NewField("id", s.id.String()),
+			xlog.NewField("blockStart", newBlock.StartTime()),
+			xlog.NewField("err", err.Error()),
+		).Errorf("error trying to drain series buffer")
+		scope.Counter("buffer-drain-error").Inc(1)
+	}
+	iOpts.MetricsScope().Counter("buffer-drained").Inc(1) // Debug
```

Comment:
Do you wan to remove this (just debug)?

Reply:
I actually think this is a useful metric, or do you mean the literally the debug comment?

Reply:
I was just trying to say this metric is mostly useful for debugging

Reply:
Ok removed the debug metric, but kept the error case one in because this is a really bad error case if its happening. That said, that codepath should never fire so allocations will be ok